### PR TITLE
Add agent ip postgre sql whitelist

### PR DESCRIPTION
**Pull request by smneal**
### Context

The Integration Tests task requires a connection to the PostgreSQL database. The hosted agent IP does not appear to be included in the "Allow access to Azure services" setting so we need to add (and remove) it for each release.

### Changes proposed in this pull request

- Add script to add the hosted agents IP to the PostgreSQL firewall whitelist
- Add script to remove the IP after

### Guidance to review

### CODE REVIEW

File: azure/scripts/Add-PostgreSQLFirewallRule.ps1 [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    if ($PSBoundParameters.ContainsKey("IpAddresses")) {
+        foreach ($Ip in $IpAddresses) {
+            $ResourceName = ("$ServerName/$FirewallRuleName-$Ip").Replace(".", "_")
+
+            New-PostgreSQLFirewallRule $ResourceName $Ip
+        }
+    }
+    else {
+        # Get the IP used by the current process/server in this instance our hosted agent
+        $ProcessIP = Invoke-RestMethod https://ipinfo.io/json | Select-Object -ExpandProperty ip
```
Comment (defong) [inline azure/scripts/Add-PostgreSQLFirewallRule.ps1]: https://ipinfo.io/json

nice :+1:

File: azure/scripts/Add-PostgreSQLFirewallRule.ps1 [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    .PARAMETER IpAddresses
+    Array of IP addresses
+
+    .EXAMPLE
+    $TestParams = @{
+        ResourceGroupName = "ServerResourceGroup"
+        ServerName = $ServerName
+        FirewallRuleName = "agent-ip"
+    }
```
Comment (defong) [inline azure/scripts/Add-PostgreSQLFirewallRule.ps1]: can you slap in `IpAddresses` for completeness

File: azure/scripts/Add-PostgreSQLFirewallRule.ps1 [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    ./Add-PostgreSQLFirewallRule.ps1 @TestParams
```
Comment (defong) [inline azure/scripts/Add-PostgreSQLFirewallRule.ps1]: PS C:\> `Get-help .\Add-PostgreSQLFirewallRule.ps1 -Detailed`

@timabell this is so so so much better than your wacky bash script

File: azure/scripts/Add-PostgreSQLFirewallRule.ps1 (hunk 2 above)
Comment (smneal) [reply azure/scripts/Add-PostgreSQLFirewallRule.ps1]: Done

**Pull request merged**